lib/components-schema-v1_0_x.ts

```diff
@@ -95,29 +99,63 @@ export const componentsDefinitionSchema_v1_0_x = {
                                     options: {
                                         type: 'array',
                                         minItems: 1,
+                                        items: {
+                                            type: 'object',
+                                            properties: {
+                                                caption: {
+                                                    type: 'string',
+                                                    description: 'Label of the item',
+                                                },
+                                                value: {
+                                                    type: 'string',
+                                                    description: 'Value of the item. Omit it if the option should simply clean the property up',
+                                                },
+                                            },
+                                            additionalProperties: false,
+                                            required: ['caption'],
+                                        },
                                     },
                                 },
-                                additionalProperties: false,
                             },
                             {
+                                additionalProperties: false,
+                                required: ['type', 'value'],
                                 properties: {
                                     type: {
                                         enum: ['checkbox'],
                                         description: 'Checkbox toggling between value and no value',
                                     },
                                     value: { type: 'string' },
                                 },
-                                additionalProperties: false,
                             },
                             {
+                                additionalProperties: false,
+                                required: ['type'],
                                 properties: {
                                     type: {
                                         enum: ['text'],
                                         description: 'Text field property',
                                     },
+                                    pattern: {
+                                        type: 'string',
+                                        description: 'Value validation regexp pattern',
+                                    },
+                                    defaultValue: {
+                                        type: 'string',
+                                        description: 'Default value which is used instead of empty value',
+                                    },
+                                    unit: {
```

Comment:
Just wondering, is it a limited number of string options we support here? If so we could further restrict it?